include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -90,15 +90,8 @@ template <class DataFacadeT, class Derived> class BasicRoutingInterface
             if (new_distance < upper_bound)
             {
                 // if loops are forced, they are so at the source
-                if (new_distance >= 0 &&
-                    (!force_loop_forward || forward_heap.GetData(node).parent != node) &&
-                    (!force_loop_reverse || reverse_heap.GetData(node).parent != node))
-                {
-                    middle_node_id = node;
-                    upper_bound = new_distance;
-                }
-                else
-                {
+                if ((force_loop_forward && forward_heap.GetData(node).parent == node) ||
```

Comment:
Yes the branching logic is not equivalent. Good catch. However is the new_distance < 0 case ever triggered without forcing at least one loop?

I'm currently trying to think of a scenario to make sure this is covered.